src/Shared/Buffers.MemoryPool/SlabMemoryPool.cs

```diff
@@ -160,6 +160,18 @@ internal void Return(MemoryPoolBlock block)
             }
         }
 
+        internal void RefreshBlock(MemoryPoolSlab slab, int offset, int length)
```

Comment:
I think this should have a name or comment that makes this abundantly clear that this is only called from the MemoryPoolBlock finalizer.